src/main/java/invtweaks/container/DirectContainerManager.java

```diff
@@ -82,8 +82,8 @@ public boolean move(ContainerSection srcSection, int srcIndex, ContainerSection
 
         // Use intermediate slot if we have to swap tools, maps, etc.
         assert srcStack != null;
-        if(destStack != null && srcStack.getItem() == destStack.getItem() && (srcStack.getMaxStackSize() == 1 ||
-                srcStack.hasTagCompound() || destStack.hasTagCompound())) {
+        if(destStack != null && srcStack.isItemEqual(destStack) && srcStack.getMaxStackSize() == 1 &&
+                ((srcStack.hasTagCompound() || destStack.hasTagCompound()) && (srcStack.getTagCompound().equals(destStack.getTagCompound())))) {
```

Comment:
I think the equals check would need to be inverted, because you *can* stack things with the same tags, but this is for doing a swap of items that can't be stacked (so would have different tags)

The whole check probably needs redone a bit though.